storage/tests/integration/signed_url_test.py

```diff
@@ -33,14 +36,46 @@ async def test_gcs_signed_url(bucket_name, creds, data, headers):
 
         signed_url = await blob.get_signed_url(60, headers=headers)
 
-        resp = await session.get(signed_url, headers=headers)
+        await verify_signed_url(blob, bucket_name, data, headers, session, signed_url, storage)
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('data', ['test'])
+@pytest.mark.parametrize('headers', [
+    {},
+    {'X-Goog-ACL': 'public-read', 'Content-Type': 'text/plain'},
+])
+async def test_gcs_iam_signed_url(bucket_name, creds, data, headers):
```

Comment:
This test doesn't exactly align with our use case, because we are not passing the IAM client in production. Here we are creating it with the creds, which are actually not available for us. A better test would be to have a real SA email available in these tests (I didn't find one), and let the library create the IAM client internally